File: cz/mg/vulkan/vk/PFNvkGetRefreshCycleDurationGOOGLE.java
package cz.mg.vulkan.vk;

public class PFNvkGetRefreshCycleDurationGOOGLE extends VkFunctionPointer {
    public PFNvkGetRefreshCycleDurationGOOGLE() {
    }

    public PFNvkGetRefreshCycleDurationGOOGLE(VkMemory vkmemory) {
        super(vkmemory);
    }

    public PFNvkGetRefreshCycleDurationGOOGLE(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }


    public PFNvkGetRefreshCycleDurationGOOGLE(long value) {
        setValue(value);
    }

    public PFNvkGetRefreshCycleDurationGOOGLE(VkInstance instance) {
        super(instance, new VkString("vkGetRefreshCycleDurationGOOGLE"));
    }

    public void call(VkDevice device, VkSwapchainKHR swapchain, VkRefreshCycleDurationGOOGLE pDisplayTimingProperties, VkResult rval){
        call(getValue(), device != null ? device.getVkAddress() : VkPointer.NULL_ADDRESS, swapchain != null ? swapchain.getVkAddress() : VkPointer.NULL_ADDRESS, pDisplayTimingProperties != null ? pDisplayTimingProperties.getVkAddress() : VkPointer.NULL, rval != null ? rval.getVkAddress() : VkPointer.NULL_ADDRESS);
    }

    private static native void call(long vkaddress, long device, long swapchain, long pDisplayTimingProperties, long rval);
}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkGetRefreshCycleDurationGOOGLE.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

void Java_cz_mg_vulkan_PFNvkGetRefreshCycleDurationGOOGLE_callNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong swapchain, jlong pDisplayTimingProperties, jlong rval){
    (void)env;
    (void)jc;
    PFN_vkGetRefreshCycleDurationGOOGLE f = (PFN_vkGetRefreshCycleDurationGOOGLE)jniLongToFunctionPointer(address);
    VkResult* rvalAddress = jniLongToPointer(rval);
    *rvalAddress = f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkSwapchainKHR*)jniLongToPointer(swapchain)),
        ((VkRefreshCycleDurationGOOGLE*)jniLongToPointer(pDisplayTimingProperties))
    );
}


jint Java_cz_mg_vulkan_PFNvkGetRefreshCycleDurationGOOGLE_callSimplifiedNative(JNIEnv* env, jclass jc, jlong address, jlong device, jlong swapchain, jlong pDisplayTimingProperties){
    (void)env;
    (void)jc;
    PFN_vkGetRefreshCycleDurationGOOGLE f = (PFN_vkGetRefreshCycleDurationGOOGLE)jniLongToFunctionPointer(address);
    return f(
        *((VkDevice*)jniLongToPointer(device)),
        *((VkSwapchainKHR*)jniLongToPointer(swapchain)),
        ((VkRefreshCycleDurationGOOGLE*)jniLongToPointer(pDisplayTimingProperties))
    );
}

